modules/utils/claws/utils/lambda_utils.hpp

```diff
@@ -7,34 +7,34 @@ namespace claws
 {
   /// \brief template to help with recursive lambdas
   ///
-  /// The basic problem is that a lambda can't refer to itself.
-  /// This class takes a lambda as parameter, and it's `operator()` injects
+  /// The basic problem is that a lambda cannot refer to itself.
+  /// This class takes a lambda as parameter, and its `operator()` injects
   /// the class itself into the lambda as first parameter,
   /// thus allowing the lambda to call the first of it's parameters to call itself
-  /// This class is usable with any functor, even though the readablity gain only realy applies to lambdas
+  /// This class is usable with any functor, even though the readablity gain only really applies to lambdas
   ///
-  /// \tparam functor_type the funtor's type
+  /// \tparam functor_type the functor's type
   template<class functor_type>
   struct inject_self
   {
     functor_type functor;
 
     inject_self() = default;
     inject_self(inject_self const &) = default;
-    inject_self(inject_self &) = default;
+    inject_self(inject_self &&) = default;
     inject_self &operator=(inject_self const &) = default;
-    inject_self &operator=(inject_self &) = default;
+    inject_self &operator=(inject_self &&) = default;
 
-    inject_self(functor_type const &functor) noexcept(std::is_nothrow_copy_constructible_v<functor_type>)
-      : functor(functor)
+    constexpr inject_self(functor_type const &f) noexcept(std::is_nothrow_copy_constructible_v<functor_type>)
```

Comment:
Well, it still has less shadowing too :p

Reply:
I don't have a problem with shadowing, do you?

Reply:
If you want to avoid shadowing, I'd rather have changes the other way around, since the constructor's prototype is what the end user sees.

Reply:
It is confusing in the case of larger functions (though it **could** be OK in our case, if it wasn't triggering a bunch of unpleasant warnings).
It also has some sort of bad practice flavor : why whould two different objects have the same name, in the same scope ?

Reply:
In this case we want to take the functor, why would you name it something else? It's not that I don't understand your argument, it's that for constructors and setters, I don't think it applies. I see no gain except either rendering the API or the implementation details uglier.

Reply:
Because there is actually no point at naming it **exactly** the same as the member variable.
- it is not the same object, it will either be copied, or have its content moved to our member variable
- the type of the object is `functor_type`, which already carries the idea that it is a functor : it is like calling an integer `number`, it does not add any extra information
- the shadowing generates unnecessary warnings
- in the case of a one-line constructor, the shadowing is not harmful, however it could be in other cases, so taking the habit of avoiding that (with the help of compiler warnings) is a safer guideline
- using the full compiler feedback on our code is useful, we cannot achieve that if the said feedback is clobbered with unnecessary warnings

Reply:
Woops, the fact the member is public is clearly a mistake on my part. Might want to fix that then.
I'm ok for using `_` if t's to avoid shadowing. I just feel it's very verbose. And it should obviously be avoided as much as possible whenever it's exposed to the user. technically, the contained field is the data of the functor (which is passed through the this pointer), so it could make sense to call it `data`. Use your best judgement, it's shouldn't be public facing anyway.